lib/src/facts/resolvers/operating_system_resolver.cc

```diff
@@ -57,6 +57,23 @@ namespace facter { namespace facts { namespace resolvers {
         if (!data.release.empty()) {
             auto value = make_value<map_value>();
 
+            // When we have no major or minor, do a 'trivial' parse of
```

Comment:
Word. We probably are doing similar "split on a period" logic in a bunch of places... We should at some point see where we can de-duplicate, but for now I'll add a method to utils and use it here.